nemo_text_processing/text_normalization/en/taggers/date.py

```diff
@@ -195,7 +212,44 @@ def __init__(self, cardinal: GraphFst, deterministic: bool):
             + day_graph
         )
 
-        final_graph = (graph_mdy | graph_dmy) + pynutil.insert(" preserve_order: true")
+        final_graph = graph_mdy | graph_dmy
+        if deterministic:
+            final_graph += pynutil.insert(" preserve_order: true")
+        else:
+            final_graph += pynini.closure(pynutil.insert(" preserve_order: true"), 0, 1)
         final_graph |= graph_ymd | year_graph_standalone
+
+        if not deterministic:
+            ymd_to_mdy_graph = None
+            mdy_to_dmy_graph = None
+
+            for month in [x[0] for x in load_labels(get_abs_path("data/months/names.tsv"))]:
+                for day in [x[0] for x in load_labels(get_abs_path("data/months/days.tsv"))]:
+                    mdy_curr = (
+                        pynutil.insert("month: \"" + month + "\" day: \"" + day + "\" ")
+                        + pynini.accep('year:')
+                        + NEMO_SIGMA
+                        + pynutil.delete(" month: \"" + month + "\" day: \"" + day + "\"")
+                    )
+
+                    mdy_curr = pynini.compose(final_graph, mdy_curr)
```

Comment:
can you give example where mdy_curr output would be different from final_graph? It looks like identity to me